dask_cuda/device_host_file.py

```diff
@@ -145,4 +146,7 @@ def __iter__(self):
 
     def __delitem__(self, key):
         self.device_keys.discard(key)
-        del self.device_buffer[key]
+        with contextlib.suppress(KeyError):
+            del self.device_buffer[key]
+        with contextlib.suppress(KeyError):
+            del self.host_buffer[key]
```

Comment:
This isn't necessary, here `device_buffer` is `fast` and `host_buffer` -- being the dict in `host_func` -- is `slow`. `zict.Buffer` will delete keys from both `fast` and `slow`: https://github.com/dask/zict/blob/3a9ea886583ba9920be8ab8ecec4adde23401a1a/zict/buffer.py#L93-L99

Reply:
So I'm curious about this point. Why do we need to specify [`__setitem__`]( https://github.com/rapidsai/dask-cuda/blob/e93ca001a2e8e2679c18824e8c5f8efb3d601c0e/dask_cuda/device_host_file.py#L125 ) and [`__getitem__`]( https://github.com/rapidsai/dask-cuda/blob/e93ca001a2e8e2679c18824e8c5f8efb3d601c0e/dask_cuda/device_host_file.py#L132 ) and handle `host_buffer` in those cases and not here?

Reply:
Additionally do we actually need `__delitem__` if this is being handled for us?

Reply:
> So I'm curious about this point. Why do we need to specify [`__setitem__`](https://github.com/rapidsai/dask-cuda/blob/e93ca001a2e8e2679c18824e8c5f8efb3d601c0e/dask_cuda/device_host_file.py#L125) and [`__getitem__`](https://github.com/rapidsai/dask-cuda/blob/e93ca001a2e8e2679c18824e8c5f8efb3d601c0e/dask_cuda/device_host_file.py#L132) and handle `host_buffer` in those cases and not here?

We need it because zict doesn't know about device or host buffers based on their type. By doing this we make sure that buffers will go directly to the appropriate dict (`device_buffer` if it's a CUDA-based buffer, `host_buffer` if CPU). When we define a maximum size for each dict (based on `device_memory_limit`/`memory_limit`) zict is able to spill once the actual buffer gets to that limit, otherwise we wouldn't be able to separate and spill according to whether device or host memory buffers are at the limit. Think for example of when we do `__getitem__`, if we just get a host item and use the `device_buffer` it would work but zict would try to "unspill" it from `host_buffer` back to `device_buffer`, which we don't want to do.

> Additionally do we actually need __delitem__ if this is being handled for us?

That's a good question, I think we need it only because of `self.device_keys.discard(key)`, otherwise it would probably be fine to drop `__delitem__` entirely.